### DISCUSSION TITLE
MongoDB Change Streams Support

### DISCUSSION THREAD

**Discussion by jburkert** [mikro-orm/mikro-orm#3174] [Ideas]
I would like to use the [MongoDB Watch API](https://www.mongodb.com/docs/manual/reference/method/db.collection.watch/) inside a MikroORM project. I could not find anything related in the documentation.

Are there any plans to support this in the future?

Thank you.

**Comment by B4nan** [comment]
I don't have any particular plans, but FWIW you can do anything the native mongo driver offers, you have the access to the native mongo collections via `em.getCollection()`, so you could use the `watch` method explicitly that way.

Not sure what exactly would you expect from this, never used this feature myself, feel free to propose something specific.

**Reply by jburkert** [reply]
For a migration we need a functionality that synchronizes between the old and new database. Both databases should be usable at the same time for the whole migration period for being able to migrate the service layer on top of it gradually to the new data model.  Besides a scheduled slow full sync we are also planning to introduce a faster event based sync process that only updates the respective documents in the new database whenever a user writes to the old database. Because this sync process is only used for the migration period I think using `em.getCollection()` to access the watch functionality will be just fine. Thank you for the hint!
